### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by ninth-dev**
- Add `view_id` to `BufferContent` and `FileContent` to simplify focus_view
- Merge `add_window_views` and `add_group_views`

### CODE REVIEW

File: opentabs.py [hunk 1]
```diff
@@ -199,34 +182,11 @@ def create_file_panel_item(self, some_content):
...
-      window.focus_view(view)
-
-
-  def find_view_by_tab_name(self, tab_name):
-    for view in self.buffers:
-      if not view.file_name() and view.name() == tab_name:
-        return view
-
-    return None
+      window.focus_view(some_content)
```
Comment (ssahayam-zd) [inline opentabs.py]: How does this work? `tracked_views` has `FileContents` or `BufferContents` not a view? Does something convert it into the view?

File: opentabs.py [hunk 2]
```diff
@@ -6,7 +6,8 @@ class FileContents:
 
   project_folder_path = "[project]"
 
-  def __init__(self, file_name, short_name, folder_name):
+  def __init__(self, view_id, file_name, short_name, folder_name):
```
Comment (ssahayam-zd) [inline opentabs.py]: can we add the `view_id` to `__str__` as well?

File: opentabs.py [hunk 3]
```diff
@@ -138,48 +137,32 @@ def run(self):
...
+  def add_views(self):
     window = self.window
     folder_name = self.get_folder_name()
     groups = list(range(window.num_groups()))
     for group_index in groups:
       views = window.views_in_group(group_index)
 
       for view in views:
+
+        self.index += 1 #start at 0
```
Comment (ssahayam-zd) [inline opentabs.py]: this whole block is kind of meaningless if you have 1 group, which is why I separated out the methods for creating views in my version

File: opentabs.py [hunk 4]
```diff
@@ -112,13 +114,10 @@ def run(self):
     self.buffers = []
```
Comment (ssahayam-zd) [inline opentabs.py]: I think we can remove `buffers`

File: opentabs.py (hunk 1 above)
Comment (ninth-dev) [reply opentabs.py]: content will have `view` instead of just `view_id`.

File: opentabs.py (hunk 3 above)
Comment (ninth-dev) [reply opentabs.py]: wrapped the code in a if 

```python
if not self.has_group:
  ....
```

**Review by ssanj — APPROVED** [review]
Thanks for your contributions!

**Pull request merged**